src/tests/adding.md

```diff
@@ -149,9 +149,12 @@ exhaustive. Header commands can generally be found by browsing the
 source.
 
 * `run-rustfix` for UI tests, indicates that the test produces
-  structured suggestions, which are then applied and the final
-  source is stored in a `.fixed` file and compiled again. The final
-  compilation is required to succeed.
+  structured suggestions. The test writer should create a `.fixed`
+  file, which contains the source with the suggestions applied.
+  When the test is run, compiletest first checks that the correct
+  lint/warning is generated. Then, it applies the suggestion and
+  compares against `.fixed` (they must match). Finally, the fixed
+  source is compiled, and this compilation is required to succeed.
```

Comment:
we should probably mention that you can run `./x.py test --bless` to generate this file